client/src/LoginPage/LoginPage.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import '../index.css';
+import { authenticationRepository, accountRepository } from '@/_services';
+
+class LoginPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            username: '',
+            password: ''
+        };
+
+        this.handleUsernameChange = this.handleUsernameChange.bind(this);
+        this.handlePasswordChange = this.handlePasswordChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleUsernameChange(event) {
+        this.setState({ username: event.target.value });
+    }
+
+    handlePasswordChange(event) {
+        this.setState({ password: event.target.value });
+    }
+
+    handleSubmit(event) {
+        event.preventDefault();
+
+        authenticationRepository.login(this.state.username.split('@')[0], this.state.password)
+            .then(() => {
+                accountRepository.create()
+                .then(id => {
+                    this.props.history.push({
+                        pathname: '/Dashboard',
+                        state: { id: id }
```

Comment:
Not sure if I would put in the URL - it would be weird to have an account ID in there, wouldn't it? I think the history state is fine for this example.

Reply:
History API is not a good place for storing this info because you wont be able to open another tab (as that state parameter will be empty). So there are 2 options:
1. return this ID with auth response and store it in the session/local storage.  
2. for simplicity reasons it could be placed in URL.

Reply:
yeah, that's true. In my mind that would be fine for the scope of this challenge, but it comes with tradeoffs.